app/v2/models/responses/hateoas/HateoasMovementResponse.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.models.responses.hateoas
+
+import play.api.libs.json.JsObject
+import play.api.libs.json.Json
+import v2.models.MovementId
+import v2.models.MovementType
+import v2.models.responses.MovementResponse
+
+object HateoasMovementResponse extends HateoasResponse {
+
+  def apply(movementId: MovementId, movementResponse: MovementResponse, movementType: MovementType): JsObject = {
+    val selfLink     = if (movementType == MovementType.Arrival) arrivalUri(movementId) else departureUri(movementId)
+    val messagesLink = if (movementType == MovementType.Arrival) arrivalMessageIdsUri(movementId) else messageIdsUri(movementId, None)
```

Comment:
messageIdsUri should changed to departureMessageIdsUri?

Reply:
I'd prefer to see this as pattern match cases -- I very much doubt we'll get another `MovementType`, but if we do (transit?), then we'll get warnings.

That said, can we just provide the movement type to the uri methods and get the message codes out from there? I don't recall if `MovementType` in the API includes `urlFragment` (that might just be in other services), but we can just use that instead of if statements like this (i.e. `s"/customs/transits/movements/${movementType.urlFragment}/...`)